src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -304,6 +306,27 @@ private boolean shouldAllowEmpty(MethodTransformerContext context, Field field)
     return true;
   }
 
+  public DynamicLangApiMethodView generateDynamicLangApiMethod(MethodTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    DynamicLangApiMethodView.Builder apiMethod = DynamicLangApiMethodView.newBuilder();
+
+    if (context.getMethodConfig().isPageStreaming()) {
+      apiMethod.type(ApiMethodType.PagedFlattenedMethod);
+    } else {
+      apiMethod.type(ApiMethodType.FlattenedMethod);
+    }
+    apiMethod.apiClassName(namer.getApiWrapperClassName(context.getInterface()));
+    apiMethod.apiVariableName(namer.getApiWrapperVariableName(context.getInterface()));
+    apiMethod.initCode(
+        initCodeTransformer.generateInitCode(
+            context, context.getMethodConfig().getRequiredFields(), new SymbolTable(), null));
+
+    apiMethod.name(namer.getApiMethodName(context.getMethod()));
+    apiMethod.hasReturnValue(!ServiceMessages.s_isEmptyType(context.getMethod().getOutputType()));
+
+    return apiMethod.build();
+  }
```

Comment:
So in the end, we might use the same function for all of the dynamic languages. We should probably call out somehow that this is an incomplete method.